src/nlp_sdg/pipelines/twitter_analytics/nodes.py

```diff
@@ -2,7 +2,44 @@
 This is a boilerplate pipeline 'twitter_analytics'
 generated using Kedro 0.18.2
 """
+import pandas as pd
+from vaderSentiment.vaderSentiment import SentimentIntensityAnalyzer
 
+#this is a dummy- node
 def dummy_node(data):
     print("Twitter Analytics dummy node completed")
-    return 5
\ No newline at end of file
+    return 5
+
+def label_tweet(df: pd.DataFrame) -> pd.DataFrame:
```

Comment:
@Emmantheo @MaureenMbugua  In my opinion, it is better practice to define a class with methods instead of nested functions. The functions inside of a function cannot be used outside of it given Python's namespaces. For example, if we want to use the function 'vader_analysis' somewhere down the line, we would have to call the 'label_tweet' function first, which is not something we always might want to do. 

In this case, I would suggest to combine the 'vadersentimentanalysis' and 'vader_analysis' into one function, and then call them as you did in the  have the 'label_tweet' function.

Reply:
Seen. We will find a way around it. Thanks for the feedback.